server/room.mjs

```diff
@@ -216,6 +220,8 @@ export default class Room {
   broadcast(func, args, exceptPlayer) {
     if(func != 'mouse')
       this.trace('broadcast', { func, args, exceptPlayer: exceptPlayer?.name });
+    if(args.playerArray && args.playerArray !== [])
+      this.players = args.playerArray;
```

Comment:
This doesn't work. `this.players` is an array of player objects that are currently in the room. You can't just change that.

Looks like `broadcast` might not be what you want. Maybe loop through the players in the audio function and test their name against your array? Or do the filtering on the receiving client side? Or change `broadcast` to optionally get an array of recipient names?

Unrelated to that problem, I don't think you can do `variable !== []`. That compares the reference so it would have to be the exact same array (which it can't because `[]` is a literal so nothing less can be that). Better to do something like `!variable.length`.

Reply:
```js
  broadcast(func, args, exceptPlayer) {
    if(func != 'mouse')
      this.trace('broadcast', { func, args, exceptPlayer: exceptPlayer?.name });
    let argsObj = {};
    for(const player of this.players) {
      let includePlayer = true;
      if(func === 'audio') {
        argsObj = JSON.parse(args);
        if(argsObj.players && !argsObj.players.includes(player.name))
          includePlayer = false;
      }
      if(player != exceptPlayer && includePlayer)
        player.send(func, args);
    }
  }
```
Also, would it be better to use two `if(func === 'audio') {}` and parse only once (outside of the `for`)?